### DISCUSSION TITLE
panda3d

### DISCUSSION THREAD

**Discussion by yunguangli** [flet-dev/flet#3208] [Packages] [answered]
https://pypi.org/project/Panda3D/

**Comment by ndonkoHenri** [comment, marked as answer]
Panda3D can't be packaged for Flet mobile apps, for two independent reasons:

- Packaging: Panda3D publishes only pre-built wheels on PyPi (no `sdist`) and is built with its bespoke `makepanda` system rather than a standard backend, so Flet's iOS/Android cross-compile toolchain can't build it.
- Architecture: Panda3D is a full 3D engine — via `ShowBase` it opens and owns the OpenGL ES window, render surface, and event loop. A Flet app is a Flutter app that already owns the screen, so there's no surface for Panda3D to render into.

If your goal is a standalone 3D game, Panda3D's own Android export (`build_apps` → `.apk`) is the right tool — but that produces a Panda3D app, not a Flet one. Flet itself doesn't currently offer an embedded 3D viewport.
